### PR TITLE
Min API tutorial: rewrite to Swagger

### LINKED ISSUE #31980: Create Min API tutorial: Update all API test instruction - all versions
### Description

Update/Rewrite the VS Code and macOS versions of this tutorial to use the latest recommended API testing tool.

This has to be done for all versions, v3-8 times the VS Code and macOS versions. 

Related to #31967

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?view=aspnetcore-8.0&tabs=visual-studio-mac

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/min-web-api.md

### Document ID

ab408035-6313-a183-9940-32de00e0e977

### Article author

@wadepickett

---
[Associated WorkItem - 226926](https://dev.azure.com/msft-skilling/Content/_workitems/edit/226926)

### PR DESCRIPTION
**Pull request by wadepickett**
Fixes #31980 

Changed the VS Code / VS for Mac versions of this tutorial to use Swagger / OpenAPI instead of Postman.

Versions 6-8 for VS, VS Code and VS for Mac were updated.

The originals had a fair amount of variation between versions to account for.

Version 6 for VS was different from the other versions in that it used a web api template with OpenAPI using Swashbuckle and it used Postman.  It didn't make much sense to remove OpenAPI with Swashbuckle, only to add OpenAPI back in with NSwag to replace Postman, so I changed that version to use an Empty template as all the other versions were using.  So now everything is using the same project template.

VS for just version 8.0 was using Endpoints Explorer.  I left that in rather than switching to NSwag.

.NET 6 did not support TypedResults and MapGroup yet, so while all else is consistent, the .NET 6 version for VS, VS Code & VS for Mac skips those sections and moves right to the Section on over-posting, with the Program.cs code updated appropriately just for that.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/tutorials/min-web-api.md](https://github.com/dotnet/AspNetCore.Docs/blob/7e88c0aa71e49b78b5b89d269175641a638ca12d/aspnetcore/tutorials/min-web-api.md) | [Tutorial: Create a minimal API with ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/min-web-api?branch=pr-en-us-32076) |

### CODE REVIEW

**Comment by wadepickett** [comment]
@CamSoper,   The VS Code and VS for Mac versions for 8.0+ are rewritten/recoded to use Swagger/OpenAPI instead of Postman.  See what you think.  I will reflect those changes in a 6.0-7.0 version if you think 8.0 looks good.

File: aspnetcore/tutorials/min-web-api.md [hunk 1]
```diff
@@ -214,18 +226,47 @@ This tutorial uses [Endpoints Explorer and .http files](xref:test/http-files#use
 
 # [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
 
-This tutorial uses Postman to test the API.
+## Create API testing UI with Swagger
+
+To test the API, this tutorial utilizes the .NET package [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore), which integrates Swagger tools for generating a testing UI adhering to the OpenAPI specification:
```
Comment (CamSoper) [inline aspnetcore/tutorials/min-web-api.md:231]: I like this approach. You might also say something about this being one option among many. Maybe something like this?

```suggestion
To test the API, you can choose from many available tools. If you don't have a preferred tool, this tutorial uses utilizes the .NET package [Swashbuckle.AspNetCore](https://www.nuget.org/packages/Swashbuckle.AspNetCore), which integrates Swagger tools for generating a testing UI adhering to the OpenAPI specification:
```

**Review by CamSoper — APPROVED** [review]
I like this approach. Approving with my one suggestion.

What do you think about using these same instructions for the VS part as well? Are we just supposed to be plugging the Endpoints Explorer for VS users? I wonder if we couldn't just use the swashbuckle instructions for both, but inform the reader that they can use Endpoints Explorer if they prefer?

**Review by CamSoper — APPROVED** [review]
I like this approach. Approving with my one suggestion.

What do you think about using these same instructions for the VS part as well? Are we just supposed to be plugging the Endpoints Explorer for VS users? I wonder if we couldn't just use the swashbuckle instructions for both, but inform the reader that they can use Endpoints Explorer if they prefer?

**Comment by wadepickett** [comment]
Thanks @CamSoper! Given the changes just announced for .NET 9, I will keep your feedback in but move this over to NSwag tonight.  Given we are just using it to generate API testing pages and nothing else, the instructions and code will change a bit but not that much overall.  Then it will be easy to update this later to a .NET 9 version.

**"What do you think about using these same instructions for the VS part as well? "**
We could use Nswag or Swachbuckle in either case as you say which would cut down on the overhead of updating this topic in the future.

I had assumed the reason Endpoints Explorer was used for VS was so we had something somewhere showing how to use it. I will ask around though.

File: aspnetcore/tutorials/min-web-api.md (hunk 1 above)
Comment (wadepickett) [reply aspnetcore/tutorials/min-web-api.md]: Got this suggestion in, but I commited a rewrite for NSwag, and then realized later that I had not commited your suggestion beforehand, so I added it after.  Sorry, didn't mean to miss getting your commit in.

**Comment by wadepickett** [comment]
Thanks @CamSoper for the review, suggestions, and approval for the 8.0 version approach.

I have now updated versions 6-8 and for each of those VS, VS Code and VS for Mac versions.

@tdykstra,
See my notes at the top in the Description for details on what changed and why.  It might be easier to review from the rendered pages than reviewing the delta in code/text changes since different versions used different templates and approaches which I had to resolve to all use NSwag and keep the variations down to just one project template so it was easier and faster to update and test.

**Review by tdykstra — APPROVED** [review]
A couple of problems to address, otherwise LGTM.

File: aspnetcore/tutorials/min-web-api.md
[the remark this answers is not in the document]
Comment (wadepickett) [reply aspnetcore/tutorials/min-web-api.md:1]: Fixed.  I went through all include versions 6-8 for vs and mac.  There were a couple to change.  Good catch.

File: aspnetcore/tutorials/min-web-api.md [hunk 2]
```diff
@@ -449,7 +504,7 @@ This method is similar to the `MapPost` method, except it uses HTTP PUT. A succe
 
 This sample uses an in-memory database that must be initialized each time the app is started. There must be an item in the database before you make a PUT call. Call GET to ensure there's an item in the database before making a PUT call.
 
-Update the to-do item that has Id = 1 and set its name to `"feed fish"`.
+Update the to-do item that has `Id = 1` and set its name to `"feed fish"`.
```
[the remark this answers is not in the document]
Comment (wadepickett) [reply aspnetcore/tutorials/min-web-api.md:507]: Fixed.  really great catches, thanks!

File: aspnetcore/tutorials/min-web-api.md [hunk 3]
```diff
@@ -361,7 +413,7 @@ Test the app by calling the `GET` endpoints from a browser or by using **Endpoin
     {
       "id": 1,
       "name": "walk dog",
-      "isComplete": false
+      "isComplete": true
```
[the remark this answers is not in the document]
Comment (wadepickett) [reply aspnetcore/tutorials/min-web-api.md:416]: Fixed.  This was in the VS 8 version only but I have checked all versions for any of the feedback. Thanks, excellent you spotted this.

**Pull request merged**